app/uk/gov/hmrc/gform/LookupLoader.scala

```diff
@@ -86,16 +228,27 @@ class LookupLoader {
   private val originSellingSomething   = read("BCD-OriginSellingSomething.csv",   "id",           "en",   "cy",   mkRadioLookup)
   private val originMainPart           = read("BCD-OriginMainPart.csv",           "id",           "en",   "cy",   mkRadioLookup)
   private val originSavingsEarnings    = read("BCD-OriginSavingsEarnings.csv",    "id",           "en",   "cy",   mkRadioLookup)
+  private val origin                   = read("BCD-Origin.csv",                   "ID",           "en",   "cy",   mkAjaxLookup(ShowAll.Enabled))
+  private val country                  = readCountries("BCD-Country.csv",         "CountryCode",  "Name", "Name-cy", "KeyWords", "Priority", "Region", mkAjaxLookup(ShowAll.Disabled))
+  private val currency                 = readCurrencies("BCD-Currency.csv",       "CurrencyCode", "Name", "Name-cy", "KeyWords", "Priority", "CountryCode", mkAjaxLookup(ShowAll.Disabled))
+  private val port                     = readPorts("BCD-Port.csv",                "PortCode",     "Name", "Name",    "KeyWords", "Priority", "Region", "PortType",  mkAjaxLookup(ShowAll.Disabled))
   // format: on
 
   private def mkAutocomplete(options: LocalisedLookupOptions): Map[LangADT, AutocompleteEngine[LookupRecord]] =
     options.m.map {
       case (l, m) =>
         val engine: AutocompleteEngine[LookupRecord] = new AutocompleteEngine.Builder[LookupRecord]()
           .setIndex(new LookupAdapter[LookupRecord]())
-          .setAnalyzers(new LowerCaseTransformer(), new WordTokenizer())
+          .setAnalyzers(new LowerCaseTransformer(), new CustomWordTokenizer())
           .build()
-        m.keys.foreach(lv => engine.add(new LookupRecord(lv.label)))
+
+        m.options map {
+          case (ll, DefaultLookupInfo(_, _)) =>
+            engine.add(new LookupRecord(ll.label, LookupPriority(1), LookupKeyWords(None)))
+          case (ll, CountryLookupInfo(_, _, k, p, _))  => engine.add(new LookupRecord(ll.label, p, k))
+          case (ll, CurrencyLookupInfo(_, _, k, p, _)) => engine.add(new LookupRecord(ll.label, p, k))
+          case (ll, PortLookupInfo(_, _, k, p, _, _))  => engine.add(new LookupRecord(ll.label, p, k))
```

Comment:
As the only thing we care about here is priority and keyword, this could be defined as a trait to avoid lots of pattern matches See https://github.com/hmrc/gform-frontend/pull/969